### PR TITLE
Clean unused dependencies from BuildFiles

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

Another quick BuildFile cleaning PR in the style of many before (for example https://github.com/cms-sw/cmssw/pull/34544).

This PR cleans unnecessary includes from CMSSW BuildFiles that were recently added for 12_1_0_pre1.

#### PR validation:

CMSSW compiles.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

No backport intended.

### CODE REVIEW

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by srimanob** [comment]
@cmsbuild please test

Give it another try.

File: SimG4CMS/HGCalTestBeam/BuildFile.xml [hunk 1]
```diff
@@ -1,15 +1,8 @@
...
 <use name="FWCore/Framework"/>
 <use name="FWCore/ParameterSet"/>
-<use name="SimDataFormats/CaloTest"/>
-<use name="SimDataFormats/GeneratorProducts"/>
-<use name="SimDataFormats/HcalTestBeam"/>
-<use name="SimDataFormats/Track"/>
-<use name="SimDataFormats/Vertex"/>
 <use name="SimG4CMS/Calo"/>
 <use name="SimG4Core/Notification"/>
-<use name="SimG4Core/Watcher"/>
```
Comment (civanch) [inline SimG4CMS/HGCalTestBeam/BuildFile.xml:12]: @bsunanda , do you agree with removal of these dependencies?

**Comment by slava77** [comment]
+reconstruction

for https://github.com/cms-sw/cmssw/pull/34809	a3765c0 
- code changes in reco are in CommonTools/CandAlgos/plugins/BuildFile.xml and appear to be correct
- jenkins tests pass

**Comment by civanch** [comment]
+1

**Comment by bbilin** [comment]
+1

**Comment by yuanchao** [comment]
+1
* clean unused depts. in buildfiles (technical)
* matrix tests and unit tests passed

**Comment by jfernan2** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

Technical PRs to clean unused dependencies.

**Comment by qliphy** [comment]
kindly ping @cms-sw/l1-l2

**Comment by qliphy** [comment]
+1

**Comment by qliphy** [comment]
merge

@cms-sw/l1-l2 Let's merge this technical PR first.

**Comment by santocch** [comment]
+1

**Pull request merged**